tests/phpunit/tests/query/searchColumns.php

```diff
@@ -0,0 +1,381 @@
+<?php
+/**
+ * Testing the search columns support in `WP_Query`.
+ *
+ * @package WordPress\UnitTests
+ * @since 6.2.0
+ */
+
+/**
+ * Test cases for the search columns feature.
+ *
+ * @group query
+ * @group search
+ *
+ * @since 6.2.0
+ */
+class Tests_Query_SearchColumns extends WP_UnitTestCase {
+	/**
+	 * The post ID of the first fixture post.
+	 *
+	 * @since 6.2.0
+	 * @var int $pid1
+	 */
+	protected static $pid1;
+
+	/**
+	 * The post ID of the second fixture post.
+	 *
+	 * @since 6.2.0
+	 * @var int $pid2
+	 */
+	protected static $pid2;
+
+	/**
+	 * The post ID of the third fixture post.
+	 *
+	 * @since 6.2.0
+	 * @var int $pid3
+	 */
+	protected static $pid3;
+
+	/**
+	 * Create posts fixtures.
+	 *
+	 * @param WP_UnitTest_Factory $factory The factory instance.
+	 */
+	public static function wpSetUpBeforeClass( $factory ) {
+		self::$pid1 = self::factory()->post->create(
```

Comment:
```suggestion
		self::$pid1 = $factory->post->create(
```
`::wpSetUpBeforeClass()` accepts the `$factory` parameter, so these can use `$factory->post->create()` instead of `self::factory()`.

🔢 Applies to the other two instances below.

Reply:
`wpSetUpBeforeClass` updated to use the `$factory` instance.